code-search/code-search.ipynb

```diff
@@ -0,0 +1,609 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Created text representations might be directly used as an input to the `all-MiniLM-L6-v2` model."
+      ],
+      "metadata": {
+        "id": "O4oN7qOjTWgq"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2DFkoAUSiW0Z"
+      },
+      "outputs": [],
+      "source": [
+        "from sentence_transformers import SentenceTransformer\n",
+        "\n",
+        "nlp_model = SentenceTransformer(\"all-MiniLM-L6-v2\")\n",
+        "nlp_embeddings = nlp_model.encode(\n",
+        "    text_representations, show_progress_bar=True,\n",
+        ")\n",
+        "nlp_embeddings.shape"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "As a next step, we are going to extract all the code snippets to a separate list. This will be an input to the different model we want to use."
+      ],
+      "metadata": {
+        "id": "g_0ko4S5TbYv"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "Y067w3WKj96o"
+      },
+      "outputs": [],
+      "source": [
+        "code_snippets = [\n",
+        "    structure[\"context\"][\"snippet\"]\n",
+        "    for structure in structures\n",
+        "]\n",
+        "code_snippets[0]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "The `jina-embeddings-v2-base-code` model is available for free, but requires accepting the rules on [the model page](https://huggingface.co/jinaai/jina-embeddings-v2-base-code). Please do it first, and put the key below."
+      ],
+      "metadata": {
+        "id": "FuxsT6O0Tl-o"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "PyNh8ccmlVSe"
+      },
+      "outputs": [],
+      "source": [
+        "# You have to accept the conditions in order to be able to access Jina embedding\n",
+        "# model. Please visit https://huggingface.co/jinaai/jina-embeddings-v2-base-code\n",
+        "# to accept the rules and generate the access token in your account settings:\n",
+        "# https://huggingface.co/settings/tokens\n",
+        "\n",
+        "HF_TOKEN = \"THIS_IS_YOUR_TOKEN\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Once the token is ready, we can pass the code snippets through the second model."
+      ],
+      "metadata": {
+        "id": "2yk2CtOtTzF8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "JcPlZxZmi4Ag"
+      },
+      "outputs": [],
+      "source": [
+        "code_model = SentenceTransformer(\n",
+        "    \"jinaai/jina-embeddings-v2-base-code\",\n",
+        "    token=HF_TOKEN,\n",
+        "    trust_remote_code=True\n",
```

Comment:
Do you want to explain a bit here what `trust_remote_code=True` is doing? That might be new to quite a few folks?

Reply:
It is actually a part of the tutorial itself, but that might be a good idea to describe it quickly here. Especially due to the recent news about HF hosting some malicious models.